app/bot.go

```diff
@@ -47,34 +47,37 @@ func (a *App) EnsureBot(c request.CTX, productID string, bot *model.Bot) (string
 		return "", errors.New("passed a bot with no username")
 	}
 
-	botIDBytes, err := a.GetPluginKey(productID, botUserKey)
-	if err != nil {
-		return "", err
+	botIDBytes, pkErr := a.GetPluginKey(productID, botUserKey)
+	if pkErr != nil {
+		return "", pkErr
 	}
 
-	// If the bot has already been created, use it
+	// If the bot has already been created, check whether it still exists and use it
 	if botIDBytes != nil {
 		botID := string(botIDBytes)
 
-		// ensure existing bot is synced with what is being created
-		botPatch := &model.BotPatch{
-			Username:    &bot.Username,
-			DisplayName: &bot.DisplayName,
-			Description: &bot.Description,
-		}
+		if _, err := a.GetBot(botID, true); err == nil {
```

Comment:
Maybe add a debug message for the `err != nil` case.

Reply:
Since we are also retrieving a deleted bot user and reusing this, I am not seeing any steps here where we specifically set the `DeleteAt` property to be nil, so that it can be reactivated and re-used. 

@hanzei, please let me know if I am missing something here.

Reply:
@srkgupta / @hanzei I don't think this is necessary, as `PatchBot` calls:

https://github.com/mattermost/mattermost-server/blob/b200a078819ebe4e6467fb23a9e41460f9113f5d/server/channels/app/bot.go#L329

which sets `trustedUpdateData` to true (second arg), which in turn does not copy `DeleteAt`:

https://github.com/mattermost/mattermost-server/blob/b200a078819ebe4e6467fb23a9e41460f9113f5d/server/channels/store/sqlstore/user_store.go#L197-L200

But not entirely sure...